packages/backend/src/models/ProjectModel/ProjectModel.ts

```
@@ -931,126 +930,159 @@ export class ProjectModel {
-                        ],
-                        change,
-                    );
+                    case 'metric':
+                    case 'dimension':
+                        const entityType = `${change.entityType}s` as const;
 
-                    const patchResult = applyPatch(
-                        explore,
-                        changedEntity === undefined
-                            ? [
-                                  {
-                                      op: 'remove',
-                                      path: `/tables/${tableName}/${entityType}/${change.entityName}`,
-                                  },
-                              ]
-                            : [
-                                  {
-                                      op: 'replace',
-                                      path: `/tables/${tableName}/${entityType}/${change.entityName}`,
-                                      value: changedEntity,
-                                  },
-                              ],
-                    );
+                        if (!patchedExplore.tables[tableName]) {
+                            throw new NotExistsError(
+                                `Table "${tableName}" does not exist in explore`,
+                            );
+                        }
 
-                    patchedExplore = patchResult.newDocument;
+                        switch (change.type) {
+                            case 'create':
+                                if (
+                                    patchedExplore.tables[tableName][
+                                        entityType
+                                    ][change.entityName]
+                                ) {
+                                    throw new ForbiddenError(
+                                        `Entity "${change.entityName}" already exists in table "${tableName}" of explore`,
+                                    );
+                                }
```

I'm considering: we should silently ignore changes that we cannot apply